### DISCUSSION TITLE
v2 docs have mostly been written, almost ready to release v2 in April

### DISCUSSION THREAD

**Discussion by lestrrat** [lestrrat-go/jwx#638] [General]
I'm done with the [docs](https://github.com/lestrrat-go/jwx/tree/v2/docs). Please read the docs, [look at the examples](https://github.com/lestrrat-go/jwx/tree/v2/examples), try the code out, and report any problems! If you have any API change suggestions, I'm still open to consider them.

* I think April 2022 is as good a time as any to release v2
* I will try to release a v2-alpha next week
* During this doc writing phase I did find some inconsistencies, and those have been incorporated into the [Changes.v2 file](https://github.com/lestrrat-go/jwx/blob/v2/Changes.v2). If you checked the v2 branch a little while ago I suggest you take a look again.

**Comment by simongottschlag** [comment]
Hi @lestrrat!

Thank you for the great work you've been doing. I've been using your library for a while now and it's wonderful. 😊⭐️🥇

In my case, I've written a small middleware that's using OIDC to setup everything: https://github.com/XenitAB/go-oidc-middleware

One thing that I've been thinking of and maybe you have some thoughts around it is if we can do something with Generics in 1.18 to have a "nicer" way of extracting claims.

It would be quite nice if it was possible to tell Parse() what type it should return.

I haven't thought this fully through, but maybe you have ideas around it? If you are aiming for a release in April, 1.18 should have been out a while.

**Reply by lestrrat** [reply]
>It would be quite nice if it was possible to tell Parse() what type it should return.

Does `jwt.WithToken()` work for you?

```go
jwt.Parse(..., jwt.WithToken(myawesometoken.New()))
```

**Reply by simongottschlag** [reply]
Hi @lestrrat!

Wouldn't we need to know the exact claims when using `WithToken()` to generate it?

I was thinking (but not sure if possible) to do something like `ParseWithType[myAwsomeTokenType]()`.

**Reply by lestrrat** [reply]
not sure what you want to achieve here. please provide some (pseudo)code?

**Reply by simongottschlag** [reply]
Sorry! Next time I get a few minutes to test out 1.18 I'll see how I want to expose my API and then get back to you how I'm thinking. (With an example)
